### PR TITLE
Simplify some renderers

### PR DESCRIPTION
**Pull request by bboreham**
`ContainerImageRenderer`: Crunch a `Map` and `Reduce` into a single `Renderer` that has the same function.

`ContainerHostnameRenderer`: The second half of the render added all the same nodes, except for
those with no Docker image information, so we could show a figure for those filtered out on that basis. This just isn't worth the effort.

And remove unused parameterisation of `containerWithImageNameRenderer`

### CODE REVIEW

File: render/container.go [hunk 1]
```diff
@@ -177,20 +169,9 @@ var ContainerImageRenderer = Memoise(FilterEmpty(report.Container,
 //
 // not memoised
 var ContainerHostnameRenderer = FilterEmpty(report.Container,
-	MakeReduce(
-		MakeMap(
-			MapContainer2Hostname,
-			ContainerWithImageNameRenderer,
-		),
-		// Grab *all* the hostnames, so we can count the number which were empty
-		// for accurate stats.
```
Comment (fbarl) [inline render/container.go]: Where would these stats be displayed?

File: render/container.go [hunk 2]
```diff
@@ -177,20 +169,9 @@ var ContainerImageRenderer = Memoise(FilterEmpty(report.Container,
...
-		MakeMap(
-			MapToEmpty,
```
Comment (fbarl) [inline render/container.go]: After removing it from here, this method is not used anymore so feel free to remove it from `container.go`.

File: render/container.go [hunk 3]
```diff
@@ -155,20 +153,14 @@ func (r containerWithImageNameRenderer) Render(ctx context.Context, rpt report.R
 
 // ContainerWithImageNameRenderer is a Renderer which produces a container
 // graph where the ranks are the image names, not their IDs
-var ContainerWithImageNameRenderer = Memoise(containerWithImageNameRenderer{ContainerRenderer})
+var ContainerWithImageNameRenderer = Memoise(containerWithImageNameRenderer{})
 
-// ContainerImageRenderer is a Renderer which produces a renderable container
-// image graph by merging the container graph and the container image topology.
+// ContainerImageRenderer produces a graph where each node is a container image
+// with the original containers as children
```
Comment (fbarl) [inline render/container.go]: :+1: Thanks for clarity!

File: render/container.go (hunk 1 above)
Comment (bboreham) [reply render/container.go]: Bottom-left of the screen, like "21 nodes (119 filtered)"

**Comment by bboreham** [comment]
I am yak-shaving towards a significant performance improvement; this one piece doesn't do much but it helps to simplify the problem.

File: render/container.go (hunk 1 above)
Comment (fbarl) [reply render/container.go]: Ok cool. That information's been misleading and incorrect at times for container topologies so I'm ok seeing it removed.

Looking at the UI code, it seems it's already robust enough to not display filtered nodes count if it's not provided so probably nothing will need to be updated there :+1:

**Pull request merged**